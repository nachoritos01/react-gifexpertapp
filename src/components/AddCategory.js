import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setImputValue] = useState('');
    
    const handleImputChange = (e)=>{
        setImputValue(e.target.value);
        // console.log('handle input change llamado ');

    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        // console.log('hndlessubmit', inputValue);

        if(inputValue.trim().length > 2){

            setCategories(cats=>[ inputValue, ...cats ]);
            setImputValue('');
        }

    }

    return (
        <form onSubmit = {handleSubmit}>
            <p>{ inputValue}</p>

            <input 
                type="text"
                value = {inputValue}
                onChange = {handleImputChange}>

            </input>
        </form>
            
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
