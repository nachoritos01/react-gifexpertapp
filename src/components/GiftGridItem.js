import React from 'react';
import PropTypes from 'prop-types';

export const GiftGridItem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img alt={title} src={url}/>
            <p>{title}</p>
        </div>
    )
}


GiftGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
