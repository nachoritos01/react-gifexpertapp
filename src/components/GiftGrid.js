import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GiftGridItem } from './GiftGridItem';

export const GiftGrid = ({category}) => {

    //Desustructuracion de los datos de retorno, y renombramos el valos de data
    const {data:images, loading} = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            { loading && <p className="animate__animated animate__fadeIn">Loading</p>}
            <div className="card-grid">
                {/* Mandamos los valores al componente individualmente cap 75 */}
                { images.map((img)=>( <GiftGridItem key={img.id} {... img}/>))}
            </div>
        </>
    )
};

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired
}