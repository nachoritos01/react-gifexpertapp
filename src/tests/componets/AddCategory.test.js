import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import React from "react";
import { AddCategory } from "../../components/AddCategory";


describe('Puebas en <AddCategory/>', () => {

    const setCategories = jest.fn();
    //Sujeto de pruebas info, con u se actualiza el snapshot
    // Solo por la ayuda del autocompletador, no es necesario
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    //Siclo de vida de las pruebas
    beforeEach(()=>{
        //Limpiar la simulacion 
        jest.clearAllMocks();
        //Reseteamos el componente como esta antes
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    });

    
    test('Pruebas con de mostrarse correctamente ', () => {
        
        expect(wrapper).toMatchSnapshot();
        
    });

    test('Dede de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        // Definimos el evento change
        input.simulate('change', { target: { value }});

        expect( wrapper.find('p').text().trim()).toBe(value);
        
    });

    test('No debe de postear la informacion con submit ', () => {
        // Mandamos la funcion prevent defaul que se esta esperando
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        // Esperamos que la funcion no era llamado
        expect( setCategories).not.toHaveBeenCalled()
    });

    test('Debe de llamar el set cateries y limpiar la caja de texto', () => {
        //1 simular el inputChange
        
        const value = 'hola mundo';
        wrapper.find('input').simulate('change', { target: { value }});

        //2 Simular el sumbnit
        wrapper.find('form').simulate('submit', { preventDefault(){}});
        //3 setCAtegories debe haber llamado

        expect(setCategories).toHaveBeenCalled();

        //Esperando que debe ser llamado con una funcion
        expect( setCategories ).toHaveBeenCalledWith(expect.any(Function));

        //4 El valor del input debe de estar ''
        console.log(wrapper.find('input').prop('value'));
        expect(wrapper.find('input').prop('value')).toBe('');
        
    });
    
    
    
    
    
})
