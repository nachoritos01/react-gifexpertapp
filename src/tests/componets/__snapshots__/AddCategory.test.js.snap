// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <AddCategory/> Pruebas con de mostrarse correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
